test/startservers.py

```diff
@@ -0,0 +1,81 @@
+import atexit
+import os
+import shutil
+import signal
+import subprocess
+import tempfile
+
+
+config = os.environ.get('BOULDER_CONFIG')
+if config is None:
+	config = 'test/boulder-config.json'
+processes = []
+tempdir = tempfile.mkdtemp()
+
+
+def run(path):
+    binary = os.path.join(tempdir, os.path.basename(path))
+    goargs = '-race' if os.environ.get('GORACE') else ''
+    cmd = 'go build %s -o %s ./%s' % (goargs, binary, path)
+    print(cmd)
+    subprocess.check_call(cmd, shell=True)
+    def _ignore_sigint():
```

Comment:
This is part of ^C being the normal way to exit start.py. Without it, ^C kills the subprocesses as well as start.py, so start.py can't tell whether the subprocesses had already died by themselves before you hit ^C. It does have the side effect that sending SIGINT directly (`killall -INT wfe`) won't kill wfe any more.

Another approach would be schedule an ALRM and call waitpid instead of using `time.sleep` in `start.py`. This would avoid infecting the integration test, and make it easy for `start.py` to exit everything right away if any one server crashes, instead of staying half-alive. This sounds desirable to me, WDYT?

(Even if we just accept the SIGINT side effect this probably deserves a comment.)